### PR TITLE
Issue 1154 fix; Display cancel download button.

### PR DESCRIPTION
**Pull request by artur-kink**
First commit fixes error in issue #1154 - https://github.com/supertuxkart/stk-code/issues/1154

Second commit adds a Cancel button during addon download.

### CODE REVIEW

**Comment by hiker** [comment]
Hmm - the assert seems indeed to be wrong, what I am not sure of is why the setManageMemory was commented out. We now have a (very very minor) memory leak there. I can't remember why I commented it out - I'll have a look at this.But since the rest of the pull request is actually independent of that - @auriamg or @deveee - can this be applied?

File: src/states_screens/credits.cpp [hunk 1]
```diff
@@ -68,6 +68,12 @@ class CreditsSection
     // ------------------------------------------------------------------------
     void addSubEntry(stringw& subEntryString)
     {
+        //Add a blank entry if no entry exists for this sub entry.
+        if(m_entries.size() == 0){
```
Comment (leyyin) [inline src/states_screens/credits.cpp]: Can we have the '{' on a new line? And also a space after the 'if'.
See section about conditionals https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Conditionals.

File: src/states_screens/dialogs/addons_loading.cpp [hunk 2]
```diff
@@ -248,6 +249,11 @@ GUIEngine::EventPropagation AddonsLoading::processEvent(const std::string& event
         voteClicked();
         return GUIEngine::EVENT_BLOCK;
     }
+    else if(event_source == "cancel")
```
Comment (leyyin) [inline src/states_screens/dialogs/addons_loading.cpp]: A space after the 'if' please.

File: src/states_screens/credits.cpp [hunk 3]
```diff
@@ -68,6 +68,12 @@ class CreditsSection
...
+            stringw blank_line(L"");
```
Comment (leyyin) [inline src/states_screens/credits.cpp]: Does this need to be initialized with empty wide string?

**Comment by hiker** [comment]
I've checked the asset, and fixed a minor memory leak (2f902f7). So cancel should now work as expected. The rest is up to @auriamg and/or @deveee :)

**Comment by auriamg** [comment]
This is no longer mergeable, got conflicts in addition to the style issues mentionned above. Since escape should work, closing as outdated

**Pull request closed without merging**